crates/runtime/src/externref/gc.rs

```diff
@@ -374,29 +382,63 @@ impl VMExternRef {
     }
 
     /// Recreate a `VMExternRef` from a pointer returned from a previous call to
-    /// `as_raw`.
+    /// `{as,into}_raw`.
     ///
     /// # Safety
     ///
-    /// Unlike `clone_from_raw`, this does not increment the reference count of the
-    /// underlying data.  It is not safe to continue to use the pointer passed to this
-    /// function.
+    /// Unlike `clone_from_raw`, this does not increment the reference count of
+    /// the underlying data. It is not safe to continue to use the pointer
+    /// passed to this function.
     #[inline]
     pub unsafe fn from_raw(ptr: *mut u8) -> Option<Self> {
         Some(VMExternRef(NonNull::new(ptr)?.cast().into()))
     }
 
+    /// Create a `&VMExternRef` from a reference to a non-null pointer that was
+    /// returned from a previous call to `{as,into}_raw`.
+    ///
+    /// Does not increment or decrement the ref count.
+    ///
+    /// # Safety
+    ///
+    /// Wildly unsafe to use with anything other than the result of a previous
+    /// `{as,into}_raw` call!
+    ///
+    /// Additionally, it is your responsibility to ensure that this raw
+    /// `VMExternRef` is still valid, or else you could have use-after-free
+    /// bugs.
+    pub unsafe fn ref_from_raw<'a>(ptr: &'a NonNull<u8>) -> &'a Self {
+        std::mem::transmute(ptr)
+    }
```

Comment:
I commented below on this but I want to be sure to leave a comment on this as well before I forget. More-or-less I think we should try to remove this method, I'm not a huge fan of methods whose bodies are just `transmute`